lib/betamocks/optional_locator.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+require 'rack'
+
+module Betamocks
+  class OptionalLocator
+    attr_writer :env, :endpoint_config
+
+    def initialize(env)
+      @env = env
+      @endpoint_config = Betamocks.configuration.find_endpoint(@env)
+    end
+
+    def generate
+      matched_optional_locator = generate_optional_locator_dir(@endpoint_config[:cache_multiple_responses])
+      return '' unless matched_optional_locator
+
+      matched_optional_locator.gsub(/[^0-9a-z]/i, '')
+    end
+
+    private
+
+    def generate_optional_locator_dir(locator_config)
+      return unless locator_config
+      # Optional Locator must exist in the same location as the UID
+      location = locator_config[:uid_location].to_sym
```

Comment:
Do we need the location assignment here? Actually, are we not able to just return if location is != body?
```
return '' if locator_config[:uid_location].to_sym != :body && !locator
```

Reply:
👍 Made the function a bit more concise